library/core/functions.validation.php

```diff
@@ -270,11 +275,18 @@ function validateUsernameRegex() {
      * @return bool Returns true if the value validates or false otherwise.
      */
     function validateUsername($value) {
-        $validateUsernameRegex = validateUsernameRegex();
+        $isPartialRegex = true;
+
+        // $isPartialRegex will be set to false if Garden.User.ValidationRegexPattern is set in the config.
+        $validateUsernameRegex = validateUsernameRegex($isPartialRegex);
+
+        if ($isPartialRegex) {
+            $validateUsernameRegex = "`^({$validateUsernameRegex})?$`siu";
```

Comment:
I see you changed to backticks, but then an admin could also put backticks in their regex. This is a difficult problem without an obvious solution. I've seen a trick where a non-printable character  as a delimiter. Perhaps @DaazKu has an idea.

Reply:
I am a big fan of keeping `/` whenever possible as the regex delimiter.
That being said ```str_replace('`', '\\`', $validateUsernameRegex)``` should do the trick here.

Reply:
The user can still escape the backtick in the config though. Example:

```php
preg_match('/\`/', '`') // returns 1
```

Reply:
If you want a foolproof solution you unescape the delimiters and the escape them again. This will prevent double escaping stuff.

`str_replace('`', '\\`', str_replace('\\`', '`', $validateUsernameRegex))`